ls1intum/Artemis
# [Exam] Schedule Programming Exercise Repository Access

### Checklist

### Motivation and Context
Only allow students to access their repository after the exam start and revoke access again after individual exam end (using the working time).

**Note: the exercise unlock operation starts when the exam becomes visible.**

### Description
- Rework programming exercise scheduling classes to support multiple differently timed tasks
- Add unlock operation for exam exercises
- Prevent exam programming exercise repository access before the exam starts / is visible.

### Steps for Testing
- Create an Exam with visible, start and end date (best case: all in the future)
- Create an exercise group with a programming exercise
- Add students to the exam
- Generate student exams (maybe with individual working times) and prepare exercise start
- Observe student repository access during the complete exam/exercise lifecycle
- Work on the exercise as a student
